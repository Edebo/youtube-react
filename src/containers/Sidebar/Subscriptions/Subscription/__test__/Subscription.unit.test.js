import React from "react";
import { shallow } from "enzyme";
import { Subscription } from "../Subscription";

describe("tesing subscription snapshot", () => {
  test("rendering empty subscription", () => {
    const wrapper = shallow(<Subscription />);

    expect(wrapper).toMatchSnapshot();
  });

  test("rendering subscription when it is broadcasting", () => {
    const wrapper = shallow(<Subscription broadcasting label='udemy' />);

    expect(wrapper).toMatchSnapshot();
  });

  test("rendering subscription when non broadcasting ", () => {
    const wrapper = shallow(<Subscription amountNewVideos={4} label='udemy' />);

    expect(wrapper).toMatchSnapshot();
  });
});
