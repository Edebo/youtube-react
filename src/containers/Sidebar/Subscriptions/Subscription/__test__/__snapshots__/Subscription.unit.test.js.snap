// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tesing subscription snapshot rendering empty subscription 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
  </div>
</MenuItem>
`;

exports[`tesing subscription snapshot rendering subscription when it is broadcasting 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        udemy
      </span>
    </div>
    <Icon
      as="i"
      name="signal"
    />
  </div>
</MenuItem>
`;

exports[`tesing subscription snapshot rendering subscription when non broadcasting  1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        udemy
      </span>
    </div>
    <span
      className="new-videos-count"
    >
      4
    </span>
  </div>
</MenuItem>
`;
